scripts/test/Muon/corrections_tab_widget/corrections_model_test.py

```diff
@@ -0,0 +1,148 @@
+        self.assertEqual(self.model.run_number_strings(), ["84447-84449"])
+
+    def test_that_current_runs_returns_none_when_a_run_string_is_not_selected(self):
+        self.assertEqual(self.model.current_runs(), None)
+
+    def test_that_current_runs_returns_the_runs_list_corresponding_to_the_currently_selected_string_in_non_coadd_mode(self):
+        self._setup_for_non_coadd_mode()
+
+        self.model.set_current_run_string("84448")
+
+        self.assertEqual(self.model.current_runs(), [84448])
+
+    def test_that_current_runs_returns_the_runs_list_corresponding_to_the_currently_selected_string_in_coadd_mode(self):
+        self._setup_for_coadd_mode()
+
+        self.model.set_current_run_string("84447-84449")
+
+        self.assertEqual(self.model.current_runs(), [84447, 84448, 84449])
+
+    def test_that_set_dead_time_source_to_from_file_will_set_the_expected_data_in_the_context(self):
+        self.model.set_dead_time_source_to_from_file()
+        self.assertEqual(self.model._corrections_context.dead_time_source, "FromFile")
+        self.assertEqual(self.model._corrections_context.dead_time_table_name, None)
+        self.assertTrue(self.model.is_dead_time_source_from_data_file())
+
+    def test_that_set_dead_time_source_to_from_ads_will_set_the_expected_data_in_the_context(self):
+        table_name = "Table name"
+        self.model.set_dead_time_source_to_from_ads(table_name)
+
+        self.assertEqual(self.model._corrections_context.dead_time_source, "FromADS")
+        self.assertEqual(self.model._corrections_context.dead_time_table_name, table_name)
+        self.assertTrue(self.model.is_dead_time_source_from_workspace())
+
+    def test_that_set_dead_time_source_to_from_ads_will_set_the_expected_data_in_the_context_to_none(self):
+        self.model.set_dead_time_source_to_from_ads("None")
+
+        self.assertEqual(self.model._corrections_context.dead_time_source, None)
+        self.assertEqual(self.model._corrections_context.dead_time_table_name, None)
+        self.assertTrue(self.model.is_dead_time_source_from_none())
+
+    def test_that_set_dead_time_source_to_none_will_set_the_expected_data_in_the_context_to_none(self):
+        self.model.set_dead_time_source_to_from_ads("Table name")
+        self.assertTrue(self.model._corrections_context.dead_time_source is not None)
+        self.assertTrue(self.model._corrections_context.dead_time_table_name is not None)
+
+        self.model.set_dead_time_source_to_none()
+
+        self.assertEqual(self.model._corrections_context.dead_time_source, None)
+        self.assertEqual(self.model._corrections_context.dead_time_table_name, None)
+        self.assertTrue(self.model.is_dead_time_source_from_none())
+
+    def test_that_dead_times_average_will_return_the_expected_dead_time_average(self):
+        dead_time_table_name = "dead_time_table"
+        LoadMuonNexus(Filename="MUSR00022725.nxs", OutputWorkspace="output_ws", DeadTimeTable=dead_time_table_name)
```

Comment:
this is slow. Could we not just make a fake dead time table?

Reply:
The code used in the model might not get the correct average, min and max from the table if it uses the wrong current run number, or mode (i.e. "FromADS" instead of "FromFile")